CONTRIBUTING.md

```diff
@@ -1,7 +1,73 @@
+# Rationale
+
+Contributions are always welcome. We accept [bug fixes](#bugs),
+[proposals and new ideas](#proposal), or simply [editorial
+improvements](#editorial-improvements).
+
+Please not that by submitting a pull request you are implicitly
+adhering to the [contributions greement](#contributions-agreement)
+
+## Bugs
+
+You can report a bug by creating an
+[issue](https://github.com/ARM-software/acle/issues). You cal also fix
+it yourself with a [pull
+request](https://github.com/ARM-software/acle/pulls).
+
+## Proposals
+
+We are open to any proposal, including the introduction of new sets of
+intrinsics. We _do not_ require to assign copyright to Arm for the
+introduction of new parts of the specifications. The original
+contributor will retain the copyright.
+
+For adding new intrinsics or preprocessor macros, we recommend
+presenting them in a [pull
+request](https://github.com/ARM-software/acle/pulls), together with
+the following information:
+
+1. A rationale covering why these intrinsics or macros are needed.
+2. A design document to keep track of the reasoning that made the
+   proposal reach its current state.
+3. Some consensus gathering in the ecosystem.
+
+Please notice that coming up with this information is not a
+requirement for submittin proposals. Contributors are trusted to use
+their judgment in deciding whether or not the proposal needs such
+data. Provinding the data is just a recommendation that will make it
+easier for new ideas to be discussed and possibly accepted, especially
+for changes of great impact.
+
+## Editorial improvements
+
+You always check the outcome of your changes by running the script
+`tools/generate-pdfs.sh`. Please notice that changes that touch the
+intrinsic database in `tools/intrinsic_db` will need to be reflected
+in the correspondent specification file by re-running the script
+`tools/generate-intrinsics-specs.sh`.
+
+These scripts are run in the CI configuration of the projects - the
+PDFs are produced as artifact and can always be downloaded in the
```

Comment:
```suggestion
PDFs are generated automatically in response to a pull request, and can be downloaded in the
```
(the world 'artifact' here may be jargon unknown to someone who doesn't already know about actions/how this is wired together, so my edit started by removing that).